File: pinyougou-parent/pinyougou-interface/src/main/java/cn/itcast/core/service/OrderService.java
package cn.itcast.core.service;

import cn.itcast.core.pojo.order.Order;
import cn.itcast.core.pojo.order.OrderItem;
import entity.PageResult;
import vo.OrderVo;

import java.util.Date;
import java.util.List;
import java.util.Map;

import entity.PageResult;
import vo.OrderVo;
import vo.PageBean;

public interface OrderService {
    void add(Order order);

	List<OrderItem> findAll();
	List<OrderVo> findAll(String name);

	PageResult findPage(Integer pageNo, Integer pageSize,String name);

	PageResult search(Integer page, Integer rows, String name, Order order,String searchDate);
	PageResult findPage(Integer pageNo, Integer pageSize);

    PageBean<OrderVo> findOrderList(Integer pageNum, Integer pageSize, String name);

	PageResult searchSta(Integer page, Integer rows, String name, Order order, String searchDate);

	Map<String,List> zheXianTu(String name);

	void changeStatus(Long id);

	PageResult searchStatistics(Integer page, Integer rows, String name, Date startDate, Date endDate);
}

File: pinyougou-parent/pinyougou-web-shop/src/main/webapp/js/service/orderService.js
//服务层
app.service('orderService', function ($http) {

    //读取列表数据绑定到表单中
    this.findAll = function () {
        return $http.get('../order/findAll.do');
    }
    //分页
    this.findPage = function (page, rows) {
        return $http.get('../order/findPage.do?page=' + page + '&rows=' + rows);
    }
    //查询实体
    this.findOne = function (id) {
        return $http.get('../order/findOne.do?id=' + id);
    }

    //搜索
    this.search = function (page, rows, searchDate,searchEntity) {
        return $http.post('../order/search.do?page=' + page + "&rows=" + rows+"&searchDate="+searchDate, searchEntity);
    }

    //更改状态  订单发货
    this.changeStatus = function (id) {
        return $http.get('../order/changeStatus.do?id='+id);
    }
});
